lib/datasource/crate/index.ts

```diff
@@ -171,17 +171,45 @@ async function fetchRegistryInfo(
 
     const cacheKey = `crate-datasource/registry-clone-path/${registryUrl}`;
 
-    let clonePath: string = memCache.get(cacheKey);
-    if (!clonePath) {
+    // We need to ensure we don't run `git clone` in parallel. Therefore we store
+    // a promise of the running operation in the mem cache, which in the end resolves
+    // to the file path of the cloned repository.
+
+    const clonePathPromise: Promise<string> | null = memCache.get(cacheKey);
+    let clonePath: string;
+
+    let cloneRepo = false;
+    if (clonePathPromise === null || clonePathPromise === undefined) {
+      cloneRepo = true;
+    } else {
+      clonePath = await clonePathPromise;
+      cloneRepo = !clonePath;
```

Comment:
Is there the possibility that (a) the clonePathPromise returns a falsey response, and (b) we'd want to clone again? Otherwise why not just `false` instead of `!clonePath`?

Reply:
Hey!

Yes, that's possible should the git clone fail for some reason. To not keep invalid state cached (e.g. clone failed, but we assume `clonePath` is valid), the promise in the memory cache will then resolve to null: 

https://github.com/renovatebot/renovate/pull/8985/files#diff-cc5ce1cdbe8a74cabc5f2c35f60d444edf0aaf35d9b59e210d6a8f3bb37a12afR200

The next dependency will then potentially retry the git clone or fail, too.

Reply:
What's the chance of:
- The same problem causes the clone to fail over and over, vs
- The problem is miraculously solved an instant later, despite running the exact same command?

I think the first is much more likely, and we're better to bail out and not retry